src/textual/driver.py

```diff
@@ -44,29 +44,37 @@ def process_event(self, event: events.Event) -> None:
         """Performs some additional processing of events."""
         if isinstance(event, events.MouseDown):
             self._mouse_down_time = event.time
+            if event.button:
+                self._down_buttons.append(event.button)
+        elif isinstance(event, events.MouseUp):
+            if event.button:
+                self._down_buttons.remove(event.button)
         elif isinstance(event, events.MouseMove):
-            if event.button and not self._dragging:
-                self._dragging = True
-                self._dragging_button = event.button
-            elif self._dragging and self._dragging_button != event.button:
-                # Artificially generate a MouseUp event when we stop "dragging"
-                self.send_event(
-                    MouseUp(
-                        x=event.x,
-                        y=event.y,
-                        delta_x=event.delta_x,
-                        delta_y=event.delta_y,
-                        button=self._dragging_button,
-                        shift=event.shift,
-                        meta=event.meta,
-                        ctrl=event.ctrl,
-                        screen_x=event.screen_x,
-                        screen_y=event.screen_y,
-                        style=event.style,
+            if (
+                self._down_buttons
+                and not event.button
+                and self._last_move_event is not None
+            ):
+                buttons = list(dict.fromkeys(self._down_buttons).keys())
```

Comment:
Isn't this the same as `list(self._down_buttons)`?

Reply:
Nope, it will additionally unique-ify them. Wether that is strictly necessary, I'm not certain.

Reply:
Ah, right. In that case, why not just `buttons = set(self._down_buttons)` and then iterate over the `set`?
Or do we _need_ to preserve order?

It makes a bit more sense, it is probably faster, and doesn't require a comment explaining the hoops you are going over.
Otherwise, I think this warrants a comment.

Depending on what you reply to this comment, I don't mind PR'ing the change.

Reply:
It is intended to preserve order. So that any button ups, arrive in the same order as they went down. Feel free to add a comment.